flink-connector-hbase-2.2/src/test/java/org/apache/flink/connector/hbase2/HBaseConnectorITCase.java

```diff
@@ -384,11 +383,13 @@ public void testTableSourceSinkWithDDL() throws Exception {
 
         TableResult tableResult3 = batchEnv.executeSql(query);
 
-        List<String> result =
-                Lists.newArrayList(tableResult3.collect()).stream()
-                        .map(Row::toString)
-                        .sorted()
-                        .collect(Collectors.toList());
+        List<String> result = new ArrayList<>();
+        for (CloseableIterator<Row> it = tableResult3.collect(); it.hasNext(); ) {
+            Row row = it.next();
+            result.add(row.toString());
+        }
+        Collections.sort(result);
+        result = new ArrayList<>(result);
```

Comment:
I think there is no need for a new arrayList, we can just reuse `result`

Anyway I would suggest to use java's `StreamSupport`, it will allow to minimize amount of changes 
the whole code block could look like  
```suggestion
        List<String> result = StreamSupport.stream(
                Spliterators.spliteratorUnknownSize(tableResult3.collect(), Spliterator.ORDERED),
                        false)
                        .map(Row::toString)
                        .sorted()
                        .collect(Collectors.toList());
```

Reply:
I agree with @snuyanzin about the second deep copy. Furthermore, the most clean solution IMO would be this:
```java
List<String> result = new ArrayList<>();
tableResult3.collect().forEachRemaining(r -> result.add(r.toString()));
```

Reply:
may be i was nt clear enough
my idea was  that since current code (before this PR) is
```java
        List<String> result =
                Lists.newArrayList(collected).stream()
                        .map(Row::toString)
                        .sorted()
                        .collect(Collectors.toList());
```
then we can just replace this part `Lists.newArrayList(collected)` with `StreamSupport.stream(
                        Spliterators.spliteratorUnknownSize(collected, Spliterator.ORDERED),
                                false)`
and leave the rest untouched instead of introducing a number of loops. This will allow to minimize amount of changes

Reply:
I see, that's a fair point. IMO since we already touch the code we can try make it simpler/cleaner in the same step. Yes, that always has the risk of introducing a regression. Anyways, fine by me to leave it with the stream api.